test-graphql/client-test.js

```diff
@@ -42,4 +55,163 @@ suite('client-test', () => {
 
     assert.ok(GraphQLJSClient.prototype.isPrototypeOf(client.graphQLClient));
   });
+
+  test('it resolves with an array of products on Client#fetchAll', () => {
+    const config = new Config({
+      domain: 'multiple-products.myshopify.com',
+      storefrontAccessToken: 'abc123'
+    });
+
+    const client = new Client(config);
+
+    fetchMock.post('https://multiple-products.myshopify.com/api/graphql', multipleProductsFixture);
+
+    return client.fetchAllProducts().then((products) => {
+      assert.ok(Array.isArray(products), 'products is an array');
+      assert.equal(products.length, 2, 'there are two products');
+
+      const passedArgs = fetchMock.lastCall('https://multiple-products.myshopify.com/api/graphql');
+
+      const passedQuery = JSON.parse(passedArgs[1].body).query;
+
+      assert.deepEqual(tokens(passedQuery.toString()), tokens(
```

Comment:
We're thinking that asserting the generated query isn't too useful/contextual to JS Buy SDK. What's more important here is asserting things about the response like the models and properties.

It's usually better to avoid tying tests to internals of a dependency as well. graphql-js-client might change something about these generated queries.